### PR TITLE
Nearest node fixes

### PR DESCRIPTION
**Pull request by permcody**
- Logic fixes in MultiAppNearestNode Transfer related to bounding box checks
- New capability to read values _from_ elements and store on either nodes or elements.

### CODE REVIEW

File: framework/src/transfers/MultiAppNearestNodeTransfer.C [hunk 1]
```diff
@@ -159,23 +164,43 @@ MultiAppNearestNodeTransfer::execute()
...
           }
         }
+
+        // By the time we get to here, we should have found at least
+        // one candidate BoundingBox for every node in the
+        // target_local_nodes array that has dofs for the current
+        // variable in the current System.
+        for (const auto & node : target_local_nodes)
+          if (node->n_dofs(sys_num, var_num) && !local_nodes_found.count(node))
+            mooseError("No candidate BoundingBoxes found for node ",
```
Comment (andrsd) [inline framework/src/transfers/MultiAppNearestNodeTransfer.C]: Needs `name()` here. When you have multiple transfers, you want to know which one failed.

File: framework/src/transfers/MultiAppNearestNodeTransfer.C [hunk 2]
```diff
@@ -202,16 +227,27 @@ MultiAppNearestNodeTransfer::execute()
...
             }
           }
         }
+
+        // Verify that we found at least one candidate bounding
+        // box for each local element with dofs for the current
+        // variable in the current System.
+        for (auto & elem : as_range(to_mesh->local_elements_begin(), to_mesh->local_elements_end()))
+          if (elem->n_dofs(sys_num, var_num) && !local_elems_found.count(elem))
+            mooseError("No candidate BoundingBoxes found for Elem ",
```
Comment (andrsd) [inline framework/src/transfers/MultiAppNearestNodeTransfer.C]: Needs `name()` here.

File: test/tests/transfers/multiapp_nearest_node_transfer/tests [hunk 3]
```diff
@@ -3,49 +3,73 @@
     type = 'Exodiff'
     input = 'tosub_master.i'
     exodiff = 'tosub_master_out_sub0.e'
+    design = 'transfers/MultiAppNearestNodeTransfer.md'
+    issues = '#1866'
+    requirement = "The NearestNodeTransfer shall transfer both nodal and elemental fields to a SubApp from the Master App."
```
Comment (andrsd) [inline test/tests/transfers/multiapp_nearest_node_transfer/tests]: Design inside requirement. Should say "The system shall...". `NearestNodeTransfer` is a product of design.

Also below.

File: framework/src/transfers/MultiAppNearestNodeTransfer.C [hunk 4]
```diff
@@ -189,10 +189,8 @@ MultiAppNearestNodeTransfer::execute()
         // variable in the current System.
         for (const auto & node : target_local_nodes)
           if (node->n_dofs(sys_num, var_num) && !local_nodes_found.count(node))
-            mooseError("No candidate BoundingBoxes found for node ",
-                       node->id(),
-                       " at position ",
-                       static_cast<Point &>(*node));
+            mooseError(
+                "No candidate BoundingBoxes found for node ", node->id(), " at position ", *node);
```
Comment (andrsd) [inline framework/src/transfers/MultiAppNearestNodeTransfer.C]: Needs `name()` here.

**Review by andrsd — CHANGES_REQUESTED** [review]

File: framework/src/transfers/MultiAppNearestNodeTransfer.C (hunk 1 above)
Comment (permcody) [reply framework/src/transfers/MultiAppNearestNodeTransfer.C]: Fixed

File: test/tests/transfers/multiapp_nearest_node_transfer/tests (hunk 3 above)
Comment (permcody) [reply test/tests/transfers/multiapp_nearest_node_transfer/tests]: Fixed

**Review by andrsd — APPROVED** [review]

**Pull request merged**